src/miral/window_specification.cpp

```diff
@@ -71,7 +71,7 @@ miral::WindowSpecification::Self::Self(mir::shell::SurfaceSpecification const& s
     pixel_format(spec.pixel_format),
     buffer_usage(),
     name(spec.name),
-    output_id(),
+    output_id(spec.output_id),
```

Comment:
This is unnecessary (see line 133) and wrong.

I was initially surprised it complies as it constructs a `optional_value<int>` with a `optional_value<graphics::DisplayConfigurationOutputId>`, but instead of taking the value of `spec.output_id.value().as_value()` it appears to take the value of `spec.output_id.operator bool()` and sets `output_id` to `0` or `1` accordingly.

There's clearly a gap in our test coverage as nothing breaks.

Reply:
Interesting :) The initializer syntax really confuses me about C++ but I can see what you mean now when you describe what happens.

What I could see from Rust (via the generated bindings so it might not be Mirs fault), output id was always set to 0 instead of not having a value so some gap in the test coverage around output_id seems very possible.